lib/model/om/BaseActor.php

```diff
@@ -641,12 +641,42 @@ public function hasChildren()
 
   public function addAncestorsCriteria(Criteria $criteria)
   {
-    return $criteria->add(QubitActor::LFT, $this->lft, Criteria::LESS_THAN)->add(QubitActor::RGT, $this->rgt, Criteria::GREATER_THAN);
+    if (isset($this->parentId))
```

Comment:
If `!isset($this->parentId)` then the object does not have any ancestors by definition, in which case I think `return $this` (the self node) is sufficient?